src/events/broker.py

```diff
@@ -94,6 +94,7 @@ def __init__(self, charm) -> None:
 
         self.provider = KafkaProvider(self)
         self.oauth = OAuthHandler(self)
+        self.kraft = KRaftHandler(self)
```

Comment:
**question** it may be a naive question, but don't we want to enable this based on the if statement above? I would have logically thought that it is either ZooKeeper or Kraft...

Reply:
We could enable it based on `ClusterState.kraft_mode` which includes condition for peer_cluster role, but there's no harm in enabling it globally because of checks existent in controller handlers.

Reply:
> globally because of checks existent in controller handlers.

Uhm, my preference here would be to have "general" checks in another "general" handler, and the custom ones related to controllers in the `KraftHandler` handler, which is run only on kraft mode. But double check with the other engineers, this is non-blocking. My only point - from a bit of an outsider perspective - is that the code above looks slightly counter intuitive to me